jbpm-test/src/test/java/org/jbpm/test/ProcessMain.java

```diff
@@ -48,12 +64,44 @@ private static KieBase readKnowledgeBase() throws Exception {
 		return kbuilder.newKieBase();
 	}
 	
-	private static void startUp() {
-		JBPMHelper.startH2Server();
-		JBPMHelper.setupDataSource();
-		// please comment this line if you already have the task service running,
-		// for example when running the jbpm-installer
-		JBPMHelper.startTaskService();
-	}
-	
-}
+    public static StatefulKnowledgeSession newStatefulKnowledgeSession(KieBase kbase) {
+    	RuntimeEnvironmentBuilder builder = null;
+    	if ( usePersistence ) {
+			Properties properties = new Properties();
+			properties.put("driverClassName", "org.h2.Driver");
+			properties.put("className", "org.h2.jdbcx.JdbcDataSource");
+			properties.put("user", "sa");
+			properties.put("password", "");
+			properties.put("url", "jdbc:h2:tcp://localhost/~/jbpm-db");
+			properties.put("datasourceName", "jdbc/jbpm-ds");
+			PersistenceUtil.setupPoolingDataSource(properties);
+		    Map<String, String> map = new HashMap<String, String>();
+		    map.put("hibernate.dialect", "org.hibernate.dialect.H2Dialect");
+	        EntityManagerFactory emf = Persistence.createEntityManagerFactory(properties.getProperty("persistence.persistenceunit.name", "org.jbpm.persistence.jpa"), map);            	            
+	        builder = RuntimeEnvironmentBuilder.Factory.get()
+                .newDefaultBuilder()
+                .entityManagerFactory(emf)                
+                .addEnvironmentEntry(EnvironmentName.TRANSACTION_MANAGER, com.arjuna.ats.jta.TransactionManager.transactionManager());
+    	} else {
+            builder = RuntimeEnvironmentBuilder.Factory.get()
+                .newDefaultInMemoryBuilder();

```

Comment:
I see that in droolsjbpm-tools you did
```
DeploymentDescriptor descriptor = 
		new DeploymentDescriptorManager().getDefaultDescriptor().getBuilder().auditMode(AuditMode.NONE).get();	
            builder.addEnvironmentEntry("KieDeploymentDescriptor", descriptor);
```

Which is triggered if `usePersistence == false`, but this is always true. So I reckon it is for users to manually change in the code. But anyway it probably needs to be added here as well.

Reply:
Changes made. I opened a new Jira ticket because it seems unseemly to have to set AuditMode to NONE for a memory Runtime. Removed datasource.properties because the default used by PersistenceUtil is sufficient.

Reply:
I think that the reason is you can have only auditing turned on - for example to store events from engine, but engine itself won't use persistence for persisting process instances etc. But maybe it is really not intended.

Reply:
Yea, the DeploymentDescriptor has AuditMode.JPA by default and the code `RuntimeEnvironmentBuilder.Factory.get() .newDefaultInMemoryBuilder();` doesn't try to change it. I think it would make the most sense to have that above routine or whatever is core to creating an in memory runtime environment set AuditMode.NONE but I didn't want to do it as part of this Jira ticket.